package.json

```diff
@@ -11,15 +11,12 @@
     "postcompile": "node scripts/postcompile.mjs",
     "watch": "tsc --build tsconfig.esm.json --watch",
     "install-with-npm-8.5": "npm i -g npm@^8.5.0 && npm i",
-    "pretest": "tsc --build tsconfig.json",
+    "pretest": "npm run compile",
```

Comment:
I don't think this is right — `pretest` builds tsconfig.json which includes tests, whereas `compile` doesn't build tests.

I don't know if the right answer here is to just always build everything (even when doing a release) or to go back to what we had before.

I think my suggestion is roughly something like

```
  "pretsc": "node scripts/pretsc.mjs",
  "tsc": "tsc",
  "compile": "npm run tsc -- --build tsconfig.build.json",
```

or something

Reply:
OK we chatted about this separately.

My understanding is there’s two things we want to get out of TSC w.r.t. tests:
- We want to make sure syntax/type errors in the tests stop the tests from running.  At least at some point in the past that wasn’t the case (diagnostics setting?) but maybe we fixed that
- We want to ensure that imports get the latest version of the code; I think that some imports (of non-test files in the current package? of files in different sub-packages ie tsc projects?) were not always doing that but were loading JS/DTS out of dist instead.

The first thing no longer seems to be an issue — we have re-enabled `diagnostics: true`.

The second thing only means that we need to make sure to build projects that aren't `noEmit: true` first, so there's no need to run tsc against the `__tests__` files which don't generate output.

Reply:
My understanding of the top-level tsconfig is that it makes vscode happy in test files too. I don’t think we ever need to run tsc on test files thanks to `ts-jest`. Everything seems to run as expected after my change and that matches my expectation.

The `.build` branch of our overall tsconfig does build the integration testsuite since we publish it (though it’s also in the `.test` branch for some reason…can probably remove). My understanding of the `.test` branch is to cover all files in `_tests_` folders to give us editor typescript happiness in those files.

In fact, it’s plausible that with the recent elimination of jest global types, that test files could roll up to our standard tsconfig and just get excluded from the build. I'll explore this and open a separate PR.
(now that we do @jest/globals stuff and got rid of the "jest" declaration in the tsconfig)